### PR TITLE
Define a hook function to run every time we absorb a block to the ledger

### PR DESCRIPTION
**Pull request by Vagabond**
The various places we absorbed to the ledger were getting fragmented, so
this hook implements the most complete set of things to do when
absorbing to the chain.

### CODE REVIEW

File: src/blockchain.erl [hunk 1]
```diff
@@ -1125,6 +1102,9 @@ reset_ledger(Height,
                                       ok = blockchain_txn:absorb_and_commit(Block, CAcc,  fun() -> ok end,
                                                                             blockchain_block:is_rescue_block(Block))
                               end,
+                              {ok, Hash} = blockchain_block:hash_block(Block),
+                              %% can only get here if the absorb suceeded
+                              run_absorb_block_hooks(true, Hash, CAcc),
```
Comment (evanmcc) [inline src/blockchain.erl:1107]: the current reset doesn't notify, but I suppose that it should, so we can use it on ETL.

File: src/blockchain.erl [hunk 2]
```diff
@@ -1724,13 +1704,7 @@ resync_fun(ChainHeight, LedgerHeight, Blockchain) ->
...
                                               lager:info("absorbing block ~p", [blockchain_block:height(Block)]),
                                               ok = blockchain_txn:unvalidated_absorb_and_commit(Block, Blockchain, fun() -> ok end, blockchain_block:is_rescue_block(Block)),
-                                              case blockchain_ledger_v1:new_snapshot(blockchain:ledger(Blockchain)) of
-                                                  {error, Reason}=Error ->
-                                                      lager:error("Error creating snapshot, Reason: ~p", [Reason]),
-                                                      Error;
-                                                  {ok, NewLedger} ->
-                                                      ok = blockchain_worker:notify({add_block, Hash, true, NewLedger})
-                                              end
+                                              run_absorb_block_hooks(true, Hashm Blockchain)
```
Comment (evanmcc) [inline src/blockchain.erl:1707]: typo, this won't compile

Comment (evanmcc) [reply src/blockchain.erl:1707]: also I guess we lose the error here but maybe we don't care?

Comment (Vagabond) [reply src/blockchain.erl:1707]: sorry, brain was not fully engaged

File: src/blockchain.erl (hunk 1 above)
Comment (Vagabond) [reply src/blockchain.erl:1108]: yeah, I think we should just always fire the hook whenever we add a block to the ledger.

**Pull request merged**